luarules/gadgets/cmd_factory_stop_production.lua

```diff
@@ -46,41 +47,62 @@ end
 --------------------------------------------------------------------------------
 
 local stopProductionCmdDesc = {
-	id      = CMD_STOP_PRODUCTION,
-	type    = CMDTYPE.ICON,
-	name    = 'Stop Production',
-	action  = 'stopproduction',
-	cursor  = 'Stop', -- Probably does nothing
-	tooltip = 'Stop Production: Clear factory production queue.',
+	id = CMD_STOP_PRODUCTION,
+	type = CMDTYPE.ICON,
+	name = "Stop Production",
+	action = "stopproduction",
+	cursor = "Stop", -- Probably does nothing
+	tooltip = "Stop Production: Clear factory production queue.",
+	hidden = false,
+	queueing = false,
 }
 
 --------------------------------------------------------------------------------
 --------------------------------------------------------------------------------
 -- Handle the command
 
 function gadget:AllowCommand_GetWantedCommand()
-	return {[CMD_STOP_PRODUCTION] = true}
+	return { [CMD_STOP_PRODUCTION] = true }
 end
 
 function gadget:AllowCommand_GetWantedUnitDefID()
 	return isFactory
 end
 
-function gadget:AllowCommand(unitID, unitDefID, teamID, cmdID, cmdParams, cmdOptions)
-	if (cmdID ~= CMD_STOP_PRODUCTION) or (not isFactory[unitDefID]) then
+function gadget:AllowCommand(unitID, unitDefID, _, cmdID)
+	if (cmdID ~= CMD_STOP_PRODUCTION) or not isFactory[unitDefID] then
 		return true
 	end
 
 	local commands = spGetFactoryCommands(unitID, -1)
-	if not commands then
+
+	if not commands or #commands == 0 then
 		return
 	end
-	for i = 1, #commands do
+
+	-- If:
+	-- - Repeat is on: clear the full queue (we can't remove the command and make it an altenqueue without losing build progress)
+	-- - Repeat is off:
+	--   - If 1 command only, clear it
+	--   - If more than 1 command, clear all except first (allow buildprogress to finish)
+	--
+	-- Remark: We can make repeat consistent with no repeat, at the cost of
+	-- adding state to track when the unit was finished after issuing the clear
+	-- queue except first cmd and clearing, however I think this is an issue
+	-- better handled in engine (we should be able to change a cmd from non-alt
+	-- to alt)
```

Comment:
Build progress loss will be solved by https://github.com/beyond-all-reason/spring/pull/1551

Reply:
Alright, when we have that online, I'll rework this, thank you